Assets/Scripts/RedRunner/Characters/RedCharacter.cs

```diff
@@ -307,26 +313,42 @@ void Awake ()
         private void LeftEvent()
         {
             if (m_State == CharacterState.Left)
-            {
-                Jump();
-            }
-            else
-            {
+			{
+				if (!m_WallDetector.TouchingWall || GroundCheck.IsGrounded)
```

Comment:
Would hit stop me from jumping if Im already moving left, fall off a cliff, hit a wall, and the hit left again?
In other words, why do we need '!m_WallDetector.TouchingWall' here? 

(same for right below)

Reply:
`!m_WallDetector.TouchingWall` is so that you can't wall jump into a wall.  The intended behavior is wall jumping only happens when switching directions.

But yeah, it would fail that case.  Adding a check if you can double jump is a way to fix that.  But I might just refactor Jump () to incorporate these checks

Reply:
Added a third check to see if you're double jumping.